src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Ascii.cs

```diff
@@ -0,0 +1,1244 @@
+        ///
+        /// This method is not intended to pair with <see cref="Equals(ReadOnlySpan{byte}, ReadOnlySpan{char})"/>
+        /// or other overloads that operate on heterogeneous parameter types.
+        ///
+        /// The output of this routine is hardened against hash code collision attacks.
+        /// </remarks>
+        public static int GetHashCode(ReadOnlySpan<byte> value)
+            => Marvin.ComputeHash32(value, Marvin.DefaultSeed);
+
+        /// <summary>
+        /// Computes a hash code over the contents of an ASCII buffer.
+        /// </summary>
+        /// <param name="value">The buffer over which to compute the hash code.</param>
+        /// <returns>A hash code for the contents of <paramref name="value"/>.</returns>
+        /// <remarks>
+        /// If two <see cref="char"/> buffers <em>left</em> and <em>right</em> compare as equal under
+        /// <see cref="Equals(ReadOnlySpan{char}, ReadOnlySpan{char})"/>, then this method
+        /// will return identical hash codes for each buffer. The implementation of this
+        /// method is not guaranteed to match the implementation of any other GetHashCode
+        /// routine in the framework, including <see cref="GetHashCodeIgnoreCase(ReadOnlySpan{char})"/>
+        /// or <see cref="string.GetHashCode"/>.
+        ///
+        /// This method is not intended to pair with <see cref="Equals(ReadOnlySpan{byte}, ReadOnlySpan{char})"/>
+        /// or other overloads that operate on heterogeneous parameter types.
+        ///
+        /// The output of this routine is hardened against hash code collision attacks.
+        /// </remarks>
+        public static int GetHashCode(ReadOnlySpan<char> value)
+            => string.GetHashCode(value);
+
+        /// <summary>
+        /// Computes a hash code over the contents of an ASCII buffer in a case-insensitive fashion.
+        /// </summary>
+        /// <param name="value">The buffer over which to compute the hash code.</param>
+        /// <returns>A hash code for the contents of <paramref name="value"/>.</returns>
+        /// <remarks>
+        /// If two <see cref="byte"/> buffers <em>left</em> and <em>right</em> compare as equal under
+        /// <see cref="EqualsIgnoreCase(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>, then this method
+        /// will return identical hash codes for each buffer. The implementation of this
+        /// method is not guaranteed to match the implementation of any other GetHashCode
+        /// routine in the framework, including <see cref="GetHashCode(ReadOnlySpan{byte})"/>.
+        ///
+        /// This method is not intended to pair with <see cref="EqualsIgnoreCase(ReadOnlySpan{byte}, ReadOnlySpan{char})"/>
+        /// or other overloads that operate on heterogeneous parameter types.
+        ///
+        /// The output of this routine is hardened against hash code collision attacks.
+        /// </remarks>
```

Comment:
just a thought: should we add a remark saying that this method might allocate an array of bytes for very big inputs? afaik we have never done that, but imho it could provide some value to the users